### PR TITLE
Swagger implementation

### PR DESCRIPTION
**Pull request by deyvidholz**
## Migrations

NO

## Description

Add swagger for API documentation

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Steps to Test or Reproduce

Follow readme

## Produced Queries

```sql
# there are no new produced or changed queries
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,9 +59,11 @@ $ yarn test:cov
 
 To generate the project documentation use `yarn doc`, it will generate the documentation usually on http://localhost:8080.
 
+Since the app is running, you can access `/api` route to check out the API documentation.
+
 ## TODO
 
-- [ ] Add openapi documentation to get swagger working on a custom endpoint
+- [x] Add openapi documentation to get swagger working on a custom endpoint
```
Comment (lucasmonstro) [inline README.md:66]: You can remove this item.

File: src/dtos/login.dto.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
+import { ApiProperty } from '@nestjs/swagger';
+
 export class LoginDTO {
+  @ApiProperty({ example: 'john@email.com' })
   username: string;
+
+  @ApiProperty({ example: 'mypassword' })
```
Comment (lucasmonstro) [inline src/dtos/login.dto.ts:7]: I suggest adding a full description about common password formats like `numbers + letters + uppercase letters + special characters`

File: src/dtos/login.dto.ts [hunk 3]
```diff
@@ -1,4 +1,9 @@
+import { ApiProperty } from '@nestjs/swagger';
+
 export class LoginDTO {
+  @ApiProperty({ example: 'john@email.com' })
```
Comment (lucasmonstro) [inline src/dtos/login.dto.ts:4]: I suggest adding a full description about `enter a valid registered email`, etc.

File: src/controllers/login.controller.ts [hunk 4]
```diff
@@ -7,6 +8,8 @@ import { JwtResult } from '../typings/jwt.typings';
 export class LoginController {
   constructor(private loginService: LoginService) {}
 
+  @ApiTags('Login')
```
Comment (lucasmonstro) [inline src/controllers/login.controller.ts:11]: I suggest adding in top of controller:
```suggestion
@ApiTags('Login')
export class LoginController {
  constructor(private loginService: LoginService) {}
```

File: src/controllers/login.controller.ts [hunk 5]
```diff
@@ -7,6 +8,8 @@ import { JwtResult } from '../typings/jwt.typings';
...
+  @ApiBody({ type: LoginDTO })
   @Post()
   @HttpCode(HttpStatus.OK)
   login(@Body() loginDTO: LoginDTO): Promise<JwtResult> {
```
Comment (lucasmonstro) [inline src/controllers/login.controller.ts:15]: For this method can you add description about what is this method and common responses like 200, 400 and 401?

**Review by lucasmonstro — CHANGES_REQUESTED** [review]

File: src/main.ts [hunk 6]
```diff
@@ -1,8 +1,19 @@
 import { NestFactory } from '@nestjs/core';
+import { SwaggerModule, DocumentBuilder } from '@nestjs/swagger';
 import { AppModule } from './app.module';
 
 async function bootstrap() {
   const app = await NestFactory.create(AppModule);
+
```
[the remark this answers is not in the document]
Comment (deyvidholz) [reply src/main.ts]: Resolved on 32735681dffcc15864113ab25766ec7db1b1ad2b.

File: src/main.ts [hunk 7]
```diff
@@ -1,8 +1,19 @@
...
+  const config = new DocumentBuilder()
+    .setTitle('JWT Authentication')
+    .setDescription('JWT Authentication built with NestJS')
+    .setVersion('1.0')
+    .build();
+
```
[the remark this answers is not in the document]
Comment (deyvidholz) [reply src/main.ts]: Resolved on 32735681dffcc15864113ab25766ec7db1b1ad2b.

File: README.md (hunk 1 above)
Comment (deyvidholz) [reply README.md]: Resolved on 32735681dffcc15864113ab25766ec7db1b1ad2b.

File: src/dtos/login.dto.ts (hunk 2 above)
Comment (deyvidholz) [reply src/dtos/login.dto.ts]: Resolved on 52765c6574c1c7fe913821432d8716b1ab659b88.

File: src/dtos/login.dto.ts (hunk 3 above)
Comment (deyvidholz) [reply src/dtos/login.dto.ts]: Resolved on 52765c6574c1c7fe913821432d8716b1ab659b88.

File: src/controllers/login.controller.ts (hunk 5 above)
Comment (deyvidholz) [reply src/controllers/login.controller.ts:27]: Resolved on f7fa703f15b2a2b058a419175eccb0b8faca3f8e.

File: README.md [hunk 8]
```diff
@@ -59,9 +59,11 @@ $ yarn test:cov
 
 To generate the project documentation use `yarn doc`, it will generate the documentation usually on http://localhost:8080.
 
+Since the app is running, you can access `/api` route to check out the API documentation.
```
[the remark this answers is not in the document]
Comment (deyvidholz) [reply README.md]: Resolved on 7d853aff428456436842cdc83efa366e4f53d84b.

**Review by lucasmonstro — APPROVED** [review]
Looks good for me. Ready for merge!

**Pull request merged**